src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -149,23 +162,40 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
         // check if already checked and passed the file
         if (!ordinaryChecks.isEmpty() || !commentChecks.isEmpty()) {
             final FileContents contents = getFileContents();
-            final DetailAST rootAST = JavaParser.parse(contents);
-            if (!ordinaryChecks.isEmpty()) {
-                walk(rootAST, contents, AstState.ORDINARY);
-            }
-            if (!commentChecks.isEmpty()) {
-                final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
-                walk(astWithComments, contents, AstState.WITH_COMMENTS);
+            DetailAST rootAST = null;
+            // whether skip the procedure after parsing Java files.
+            boolean skip = false;
+
+            try {
+                rootAST = JavaParser.parse(contents);
```

Comment:
I am open to creating a Try monad for this purpose, so that we can return a value or exception and deal with the result accordingly.

Reply:
Sure, thanks for your advice. I will try to create a try monad :)

Reply:
Other maintainers might not like this, it might be good to get some feedback first

Reply:
Please better to postpone monad improvement to next PR. Let's merge functional fix first.

Reply:
@romani @nrmancuso @rnveach Could you please take a look at https://github.com/checkstyle/checkstyle/pull/14779#discussion_r1564914632. The new workaround is kind of wired to me.

Reply:
> @romani @nrmancuso @rnveach Could you please take a look at [#14779 (comment)](https://github.com/checkstyle/checkstyle/pull/14779#discussion_r1564914632). The new workaround is kind of wired to me.

Yes, I do not like this update, which is why I recommended a Try monad for to avoid mutable flags and other hacks.